types.go

```diff
@@ -908,6 +935,37 @@ type (
 		NetAmount           PurchaseUnitAmount  `json:"net_amount"`
 		TotalRefundedAmount *PurchaseUnitAmount `json:"total_refunded_amount,omitempty"`
 	}
+
+	ReferralRequest struct {
+		TrackingID    string      `json:"tracking_id"`
+		Operations    []Operation `json:"operations,omitempty"`
+		Products      []string    `json:"products,omitempty"`
+		LegalConsents []Consent   `json:"legal_consents,omitempty"`
+	}
+
+	Operation struct {
+		Operation                string             `json:"operation"`
+		APIIntegrationPreference IntegrationDetails `json:"api_integration_preference,omitempty"`
+	}
+
+	IntegrationDetails struct {
+		RestAPIIntegration RestAPIIntegration `json:"rest_api_integration"`
```

Comment:
is this always present? or do we need to omitempty

Reply:
I believe it is always present. The documentation isn't very clear.

Reply:
if the doc isnt clear can we rather omitempty as it is safer.

Reply:
```suggestion
		RestAPIIntegration *RestAPIIntegration `json:"rest_api_integration,omitempty"`
```